### PR TITLE
VersionControlSystem: Try to determine the VCS type statically first 

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

**Review by mnonnenmacher — DISMISSED** [review]

File: downloader/src/main/kotlin/VcsHost.kt [hunk 1]
```diff
@@ -304,6 +304,18 @@ enum class VcsHost(
...
+                    val query = "?version=GB"
+                    val url = projectUrl.substringBefore(query)
+                    val revision = projectUrl.substringAfter(query, "")
+
+                    VcsInfo(
+                        type = VcsType.GIT,
+                        url = url,
+                        revision = revision
+                    )
+                }
```
Comment (fviernau) [inline downloader/src/main/kotlin/VcsHost.kt:317]: Would be nice to have a test which illustrates how the TFS URLs actually look like.

**Review by fviernau — CHANGES_REQUESTED** [review]

File: downloader/src/main/kotlin/VcsHost.kt (hunk 1 above)
Comment (sschuberth) [reply downloader/src/main/kotlin/VcsHost.kt:317]: Done.

Comment (oheger-bosch) [reply downloader/src/main/kotlin/VcsHost.kt:317]: The tests should cover all the conditions under which a TFS URL is detected, also the part with ".visualstudio.com". 
Could it be that the URL contains multiple query parameters? Should the parsing of the version be made more robust, e.g. by splitting the query string at `&` characters and inspecting the key-value pairs?

Comment (sschuberth) [reply downloader/src/main/kotlin/VcsHost.kt:320]: Done.

**Pull request merged**